dashboard/app/controllers/application_controller.rb

```diff
@@ -402,10 +402,10 @@ def prevent_caching
     redirect_to lti_v1_account_linking_landing_path
   end
 
-  # Creates a stable statsig id for use of session tracking (whether the user is logged in or not)
-  # Use this session variable when you want to track the user journey when the user is not logged in.
-  protected def initialize_statsig_session
-    session[:statsig_stable_id] ||= SecureRandom.uuid
+  # Creates a statsig stable id for use of signed-out user tracking.
+  # This cookie is used by the Statsig SDK for both JS and Ruby.
+  protected def initialize_statsig_stable_id
+    cookies[:statsig_stable_id] ||= {value: SecureRandom.uuid, domain: :all, path: '/'}
   end
```

Comment:
Since we only need this ID to distinguish between events from different users, wouldn’t it make sense to use the session ID we already store in cookies? Especially since our cookie storage is already overloaded.
```ruby
cookies[environment_specific_cookie_name('_learn_session')]
```
https://github.com/code-dot-org/code-dot-org/blob/dafeb0e38218bd6149b1e89dd4cb4753f26dc4ae/dashboard/config/initializers/session_store.rb#L3-L5

Reply:
We need this ID to be set even for signed-out users (users without a session), as well as remain persistent if a signed-out user becomes a signed-in user.

Reply:
The session ID is always present, even before signing in and after signing out. The only difference is that after signing in or out, the ID will be changed.

Reply:
Yep, I realized that after I wrote this. However, we don't want it to change after the user signs in, as we won't be able to tie the user journey of a signed-out user becoming a signed-in user if that stable id changes.